drivers/source/SerialBase.cpp

```diff
@@ -287,6 +287,10 @@ SerialBase::~SerialBase()
     for (int irq = 0; irq < IrqCnt; irq++) {
         attach(nullptr, (IrqType)irq);
     }
+
+    if (_rx_enabled || _tx_enabled) {
+        serial_free(&_serial);
```

Comment:
I wonder how come this was not added in #10924 where we have added enable/disable output.